cassdegrees/ui/views.py

```diff
@@ -224,73 +223,100 @@ def manage_courses(request):
+            elif action == 'Edit':
+                id_to_edit = post_data.get('id')
+                if id_to_edit:
+                    render_properties['hide_form'] = False
+                    print("self is true" + id_to_edit)
+                    # Patch requests (editing an already existing resource only requires fields that are changed
+                    offered_sems = post_data.getlist('semesters[]')
+                    course_instance = \
+                        {
+                            'id': id_to_edit,
+                            'code': post_data.get('code'),
+                            'year': post_data.get('year'),
+                            'name': post_data.get('name'),
+                            'units': post_data.get('units'),
+                            'offeredSem1': 'semester1' in offered_sems,
+                            'offeredSem2': 'semester2' in offered_sems
+                        }
 
-        # to be implemented, currently has the sample model code
-        elif action == 'Edit':
-            id_to_edit = post_data.get('id')
-            # Patch requests (editing an already existing resource only requires fields that are changed
-            course_instance = \
-                {
-                    'text': post_data.get('text')
-                }
+                    rest_api = requests.patch(model_api_url + id_to_edit + '/', data=course_instance)
 
-            rest_api = requests.patch(model_api_url + id_to_edit + '/', data=course_instance)
+                    if rest_api.status_code == 200:
+                        render_properties['msg'] = 'Course information successfully modified!'
+                    else:
+                        render_properties['is_error'] = True
+                        render_properties['msg'] = "Failed to edit course information. Please try again."
+
+        elif request_origin == 'list.html':
+            if action == 'Edit':
+                id_to_edit = post_data.get('id')
+                if id_to_edit:
+                    render_properties['hide_form'] = False
+                    print("id_to_edit: " + id_to_edit)
+                    current_course_info = requests.get(model_api_url + id_to_edit + '/?format=json').json()
+                    current_course_info['id'] = id_to_edit
+                    render_properties['edit_course_info'] = current_course_info
+                    print("all info: " + str(current_course_info))
 
-            if rest_api.status_code == 200:
-                render_properties['msg'] = 'Course information successfully modified!'
-            else:
-                render_properties['is_error'] = True
-                render_properties['msg'] = "Failed to edit course information (unknown error). Please try again."
+                else:
+                    render_properties['is_error'] = True
+                    render_properties['hide_form'] = True
+                    render_properties['msg'] = "Choose a course to edit."
```

Comment:
Same feedback from before - rewording or at the very least an exclamation mark for an error.

Reply:
Altered message to be "Please select a course to edit!"